### DISCUSSION TITLE
Fingerprinting

### DISCUSSION THREAD

**Discussion by AuRobinson** [Jelmerro/Vieb#577] [Q&A]
# Unique Fingerprinting
## Question
Does/will Vieb provide a randomized fingerprint?

## Thoughts
Based on the results of the EFF test (cover your tracks), it appears that Vieb does not combat fingerprinting. However, the second site (AM I UNIQUE?) shows that Vieb is unique.

Are there some validity to either of these tests? Or is it the case that Vieb is built in a way that provides adequate anonymity?

## Site Resources
- [EFF Browser Test: Cover Your Tracks](https://coveryourtracks.eff.org/)
  - [Methodology](https://coveryourtracks.eff.org/about)
- [Am I Unique](https://www.amiunique.org/fingerprint)

**Comment by Jelmerro** [comment]
I think we are doing ok, not great nor fantastic, but ok, most of the overrides/fixes are in this file: https://github.com/Jelmerro/Vieb/blob/master/app/preload/privacy.js
for example:
- cpu and memory are fixed to 8
- keyboard info is blocked
- battery info is blocked
- network info is blocked
- prompt/alert/confirm are mocked so they don't interrupt
- permissions are blocked by default (that is in app/index.js mostly, although the mediadevices are in privacy.js)
- webgl gpu names are blocked

Most of the other stuff like DNT, plugin lists and encoding sets are set to chromium/chrome's default values. The reason these tests are unique is due to canvas and WebGL hashes, which I have yet to find a solution for that runs in the Electron project. As per usual, PRs are welcome.
